components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2733,55 +2742,124 @@ private void processPatchRequest(List<Patch> patchRequest, IdentityProvider idpT
+                    } else {
                         throw handleException(Response.Status.BAD_REQUEST,
-                                Constants.ErrorMessage.ERROR_CODE_INVALID_CERTIFICATE_FORMAT, null);
+                                Constants.ErrorMessage.ERROR_CODE_INVALID_INPUT, "Invalid index in 'path' attribute");
                     }
-                }
-                certificates.add(index, value);
-                idpToUpdate.setCertificate(base64Encode(StringUtils.join(certificates, "")));
-            } else if (operation == Patch.OperationEnum.REMOVE && path.matches(Constants.CERTIFICATE_PATH_REGEX) &&
-                    path.split(Constants.PATH_SEPERATOR).length == 4) {
-                List<String> certificates = new ArrayList<>();
-                int index = Integer.parseInt(path.split(Constants.PATH_SEPERATOR)[3]);
-                if (ArrayUtils.isNotEmpty(idpToUpdate.getCertificateInfoArray()) && (index >= 0) && index <
-                        idpToUpdate.getCertificateInfoArray().length) {
-                    for (CertificateInfo certInfo : idpToUpdate.getCertificateInfoArray()) {
-                        certificates.add(base64Decode(certInfo.getCertValue()));
+                    idpToUpdate.setCertificate(base64Encode(StringUtils.join(certificates, "")));
+                } else if (Constants.CERTIFICATE_JWKSURI_PATH.equals(path)) {
+
+                    IdentityProviderProperty[] propertyDTOS = idpToUpdate.getIdpProperties();
+                    List<IdentityProviderProperty> idpNewProperties = new ArrayList<>();
+                    for (IdentityProviderProperty propertyDTO : propertyDTOS) {
+                        // Add properties to new list omitting the JWKS URI property.
+                        if (!Constants.JWKS_URI.equals(propertyDTO.getName())) {
+                            idpNewProperties.add(propertyDTO);
+                        }
+                    }
+
+                    // If the sizes of original and new property lists are equal, then the JWKS URI property has not
+                    // been available.
+                    if (propertyDTOS.length == idpNewProperties.size()) {
+                        throw handleException(Response.Status.NOT_FOUND,
+                                Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_IDP,
+                                "Cannot remove JWKS URI as it does not exist");
                     }
-                    certificates.remove(index);
+
+                    idpToUpdate.setIdpProperties(idpNewProperties.toArray(new IdentityProviderProperty[0]));
                 } else {
-                    throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
-                            .ERROR_CODE_INVALID_INPUT, "Invalid index in 'path' attribute");
+                    throw handleException(Response.Status.BAD_REQUEST,
+                            Constants.ErrorMessage.ERROR_CODE_INVALID_INPUT, null);
```

Comment:
This method is too long. Better if we can refactor it by extracting out private methods

Reply:
Related to the dependent PR